Добавлено решение для шифров Цезаря и Виженера, RSA-шифрования.

annparkhomenk opened the pull request:
Реализованы функции encrypt_caesar и decrypt_caesar, encrypt_vigenere и decrypt_vigenere, а также вспомогательная функция key_word для генерации ключа в шифре Виженера.

annparkhomenk committed:
Добавлено решение для шифров Цезаря и Виженера. 

Реализованы функции encrypt_caesar и decrypt_caesar, encrypt_vigenere и decrypt_vigenere, а также вспомогательная функция key_word для генерации ключа в шифре Виженера.

annparkhomenk committed:
Реализована генерация ключа в RSA-шифровании.

Реализованы функции is_prime(), gcd(), multiplicative_inverse() и generate_keypair().

anastasiiaCher commented on caesar.py:
```diff
@@ -0,0 +1,33 @@
+    ciphertext = ""
+    for i in plaintext:
+        if i.isalpha():
+            o = ord(i)
```
Что такое `o`?

anastasiiaCher commented on caesar.py:
```diff
@@ -0,0 +1,33 @@
+    for i in plaintext:
+        if i.isalpha():
+            o = ord(i)
+            if (97 <= o <= 122 - shift) or (65 <= o <= 90 - shift):
```
Хардкодить - плохо? Что такое 97 и 122? В данном случае `or` не обязывает нас брать выражения в скобки

anastasiiaCher commented on caesar.py:
```diff
@@ -0,0 +1,33 @@
+        else:
+            plaintext += i
+
+    return plaintext
```
То же самое, что и по предыдущей функции.

anastasiiaCher commented on rsa.py:
```diff
@@ -0,0 +1,93 @@
+                break
+    else:
+        prime = False
+    return prime
```
Под логическое значение можно не вводить переменную, а сразу возвращать через `return`

anastasiiaCher commented on vigenere.py:
```diff
@@ -0,0 +1,52 @@
+def key_word(text, keyword):
```
Реализация шифра виженера не предполагает введения вспомогательных функций. Можно сделать проще

anastasiiaCher commented on vigenere.py:
```diff
@@ -0,0 +1,52 @@
+
+    key_int = key_word(plaintext, keyword)
+
+    for i in range(len(plaintext)):
```
Здесь лучше сделать так: `for i, letter in enumerate(plaintext)`, чтобы дальше не обращаться по индексу к строке

anastasiiaCher replied:
Дальше по коду замечания те же, что и по цезарю

anastasiiaCher reviewed (changes requested):
В целом с логикой и реализацией все ок, но есть ряд замечаний. Плюс Actions должны проходить

annparkhomenk committed:
Исправлены ошибки в caesar.py

annparkhomenk committed:
Исправлены ошибки в caesar.py (2)

annparkhomenk committed:
Исправлены ошибки в vigenere.py

annparkhomenk committed:
Исправлены ошибки в rsa.py

annparkhomenk commented:
Я исправила ошибки, про которые Вы написали. Насчет Actions не знаю, там в mypy пишут, что в rsa name '..' not defined, хотя все переменные определены.

annparkhomenk committed:
Перенесены изменения в файл caesar.py

Перенесены изменения в файл caesar.py, лежащий в папке homework01, из файла caesar.py, лежащего вне папки (неправильно загруженного).

annparkhomenk committed:
Перенесены изменения в vigenere.py в папке homework01

annparkhomenk committed:
Перенесены изменения в rsa.py в папку homework01

annparkhomenk committed:
Удален лишний файл caesar.py

annparkhomenk committed:
Удален лишний файл rsa.py

annparkhomenk committed:
Удален лишний файл vigenere.py

annparkhomenk committed:
Исправлены ошибки в оформлении caesar.py

annparkhomenk committed:
Исправлены ошибки в оформлении vigenere.py

annparkhomenk committed:
Исправлены ошибки в оформлении rsa.py

annparkhomenk commented:
Оказалось, что изначально я загрузила файлы не туда (не в папку homework01, а просто на ветку). Поэтому не проходили Actions. Последние коммиты -- это перенос файлов туда, куда надо, удаление ненужных файлов и исправление оформления (чтобы black проходил). Теперь Actions прошли.

anastasiiaCher reviewed (approved):
Теперь в целом все ок. Только не удаляйте больше доктесты (это то, что в тройных кавычках сразу после заголовка функции) нигде.
